### ISSUE TITLE
./geth_heco_v1_2_2 --config "config.toml"

### ISSUE THREAD

**Issue by xian9yu** [stars-labs/heco-chain#58]
Fatal: Failed to register the Ethereum service: unsupported fork ordering: sophonBlock not enabled, but arrowGlacierBlock enabled at 0

这是怎么回事, release里直接下载的 v1.2.2

**Comment by Nanyan** [comment]
Are you running a private chain? If so, you should enabled sophon hardfork in your `genesis.json` before enabling arrowGlacier hardfork.
Such as: 
```json
{
  "config": {
    "chainId": 6660001,
    "homesteadBlock": 0,
    "eip150Block": 0,
    "eip150Hash": "0x0000000000000000000000000000000000000000000000000000000000000000",
    "eip155Block": 0,
    "eip158Block": 0,
    "byzantiumBlock": 0,
    "constantinopleBlock": 0,
    "petersburgBlock": 0,
    "istanbulBlock": 0,
    "muirGlacierBlock": 0,
    "redCoastBlock": 3,
    "sophonBlock": 1000,
    "congress": {
      "period": 3,
      "epoch": 100
    }
......
  }

**Comment by Nanyan** [comment]
Or you should delete `arrowGlacierBlock` in your `genesis.json` for the first run of your chain.

**Comment by xian9yu** [comment]
> Or you should delete `arrowGlacierBlock` in your `genesis.json` for the first run of your chain.

```
"config": {
    "chainId": 12300,
    "homesteadBlock": 0,
    "eip150Block": 0,
    "eip150Hash": "0x0000000000000000000000000000000000000000000000000000000000000000",
    "eip155Block": 0,
    "eip158Block": 0,
    "byzantiumBlock": 0,
    "constantinopleBlock": 0,
    "petersburgBlock": 0,
    "istanbulBlock": 0,
    "londonBlock": 0,
    "berlinBlock": 0,
    "muirGlacierBlock": 0,
    "redCoastBlock": 0,
    "sophonBlock": 0,
    "congress": {
      "period": 3,
      "epoch": 200
    }
  }
```
还是不得行, 提示Fatal: Failed to register the Ethereum service: unsupported fork ordering: sophonBlock not enabled, but arrowGlacierBlock enabled at 0

**Comment by xian9yu** [comment]
> Or you should delete `arrowGlacierBlock` in your `genesis.json` for the first run of your chain.

老哥, 给个私链添加验证者的教程, 百度找到的教程添加不了

**Comment by zzyalbert** [comment]
The latest v1.2.2 has been updated. 

Please update the binary and hard forks config like the following. I think your issue may be resolved.

`"config": {
    "chainId": 12300,
    "homesteadBlock": 0,
    "eip150Block": 0,
    "eip150Hash": "0x0000000000000000000000000000000000000000000000000000000000000000",
    "eip155Block": 0,
    "eip158Block": 0,
    "byzantiumBlock": 0,
    "constantinopleBlock": 0,
    "petersburgBlock": 0,
    "istanbulBlock": 0,
    "londonBlock": 0,
    "berlinBlock": 0,
    "muirGlacierBlock": 0,
    "redCoastBlock": 2,
    "sophonBlock": 3,
    "congress": {
      "period": 3,
      "epoch": 200
    }
  }`

We really appreciate your report to this issue and sorry for the inconvenience.

**Comment by xian9yu** [comment]
> 

修改genesis.json之前的链就不能重新用了, 怎么直接加分叉配置的呢

**Comment by xian9yu** [comment]
> The latest v1.2.2 has been updated.
> 
> Please update the binary and hard forks config like the following. I think your issue may be resolved.
> 
> ```json
> "config": {
>     "chainId": 12300,
>     "homesteadBlock": 0,
>     "eip150Block": 0,
>     "eip150Hash": "0x0000000000000000000000000000000000000000000000000000000000000000",
>     "eip155Block": 0,
>     "eip158Block": 0,
>     "byzantiumBlock": 0,
>     "constantinopleBlock": 0,
>     "petersburgBlock": 0,
>     "istanbulBlock": 0,
>     "londonBlock": 0,
>     "berlinBlock": 0,
>     "muirGlacierBlock": 0,
>     "redCoastBlock": 2,
>     "sophonBlock": 3,
>     "congress": {
>       "period": 3,
>       "epoch": 200
>     }
>   }
> ```
> 
> We really appreciate your report to this issue and sorry for the inconvenience.

修改genesis.json之前的链就不能重新用了, 怎么直接加分叉配置的呢

**Comment by Nanyan** [comment]
the `config` content in the genesis.json file can be changed as long as the new config is compatible with the old config.

**Comment by Nanyan** [comment]
This is fixed on new released v1.2.2
